### PR TITLE
Onboarding task6

### PR DESCRIPTION
**Pull request by thalyscarrara**
## Description

implement codelabs to-do list app

## Related Issues

[task06-onboarding](https://github.com/profusion/quickstart-android-jetpack-graphql/issues/201)

https://user-images.githubusercontent.com/53837410/191854714-5fde81a1-0b7d-4de8-986d-b37ff79827e1.mp4

)

### CODE REVIEW

File: app/src/main/java/com/codelabs/basicstatecodelab/WaterCounter.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.codelabs.basicstatecodelab
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.* // ktlint-disable no-wildcard-imports
```
Comment (mari1912) [inline app/src/main/java/com/codelabs/basicstatecodelab/WaterCounter.kt]: Avoid to disable ktlint. It's here to help you

File: app/src/main/java/com/codelabs/basicstatecodelab/WaterCounter.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun StatelessCounter(count: Int, onIncrement: () -> Unit, modifier: Modifier = Modifier) {
+    Column(modifier = modifier.padding(16.dp)) {
+        if (count > 0) {
+            Text("You've had $count glasses.")
+        }
+        Button(onClick = onIncrement, Modifier.padding(top = 8.dp), enabled = count < 10) {
+            Text("Add one")
```
Comment (mari1912) [inline app/src/main/java/com/codelabs/basicstatecodelab/WaterCounter.kt]: All the strings should be in a `strings.xml` file

File: app/src/main/java/com/codelabs/basicstatecodelab/WaterCounter.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import androidx.compose.runtime.* // ktlint-disable no-wildcard-imports
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun StatelessCounter(count: Int, onIncrement: () -> Unit, modifier: Modifier = Modifier) {
+    Column(modifier = modifier.padding(16.dp)) {
+        if (count > 0) {
+            Text("You've had $count glasses.")
```
Comment (mari1912) [inline app/src/main/java/com/codelabs/basicstatecodelab/WaterCounter.kt]: ditto

File: app/src/main/java/com/codelabs/basicstatecodelab/WellnessViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        _tasks.remove(item)
+    }
+
+    fun changeTaskChecked(item: WellnessTask, checked: Boolean) =
+        tasks.find { it.id == item.id }?.let { task ->
+            task.checked = checked
+        }
+}
+
+private fun getWellnessTasks() = List(30) { i -> WellnessTask(i, "Task # $i") }
```
Comment (mari1912) [inline app/src/main/java/com/codelabs/basicstatecodelab/WellnessViewModel.kt]: ditto

File: app/src/main/java/com/codelabs/basicstatecodelab/WellnessViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        get() = _tasks
+
+    fun remove(item: WellnessTask) {
+        _tasks.remove(item)
+    }
+
+    fun changeTaskChecked(item: WellnessTask, checked: Boolean) =
+        tasks.find { it.id == item.id }?.let { task ->
+            task.checked = checked
+        }
```
Comment (joaobertacchi) [inline app/src/main/java/com/codelabs/basicstatecodelab/WellnessViewModel.kt]: Here you turned `checked` into a mutable state (as the suggested approach in the tutorial). A more-React way of doing that would be to update the whole list. Could you implement that alternative solution?

**Review by joaobertacchi — APPROVED** [review]
Pre-approving

**Review by mari1912 — APPROVED** [review]

**Pull request merged**